vars/testBranchRE.groovy

```diff
@@ -4,6 +4,28 @@
    * testBranchRE step method
    */
 
-def call() {
-    return '^[-0-9A-Za-z.]+-testing(-(tcp|ucx))?(-2.2)?$'
-}
\ No newline at end of file
+def call(String test_branch_type='') {
+    // Get a regular expression to match a testing branch name.
+    switch (test_branch_type) {
+        case 'weekly':
+            // Match a weekly test branch, e.g. weekly-testing, weekly-<release>-testing, etc.
+            // Also support older weekly-testing-<release> branch names
+            return '^weekly-([.-0-9]+)?testing(-2.2)?$'
```

Comment:
```suggestion
            return '^weekly-([.-\d]+)?testing(-2.2)?$'
```

Reply:
This didn't work in build 4, so I reverted it and it passed in build 5:
```
[2023-06-13T22:20:08.137Z] Unit testing, so exiting "Get test list" with true

startup failed:

/data/jenkins/jobs/daos-stack/jobs/pipeline-lib/branches/PR-378/builds/4/libs/9ccc08329bee55aecceb648f322a9a43c136afc6a1cbaecd11ff3422ff27945b/vars/branchTypeRE.groovy: 13: unexpected char: '\' @ line 13, column 33.

               return '^weekly-([.-\d]+)?testing(-2.2)?$'

                                   ^

1 error
```

Reply:
Jenkins (or maybe it's just groovy) seems to be funny about `\`.  I find (at times?) that I need to double escape using `\\`.

But also worth noting here per daos-stack/daos#12387, the class needs to start with a `-` if you want to match a literal `-`.